features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/RoomListPresenter.kt

```diff
@@ -173,33 +172,48 @@ class RoomListPresenter @Inject constructor(
     }
 
     @Composable
-    private fun securityBannerState(
+    private fun rememberSecurityBannerState(
         securityBannerDismissed: Boolean,
         needsSlidingSyncMigration: Boolean,
     ): State<SecurityBannerState> {
         val currentSecurityBannerDismissed by rememberUpdatedState(securityBannerDismissed)
+        val currentNeedsSlidingSyncMigration by rememberUpdatedState(needsSlidingSyncMigration)
         val recoveryState by encryptionService.recoveryStateStateFlow.collectAsState()
-        val syncState by syncService.syncState.collectAsState()
         return remember {
             derivedStateOf {
-                when {
-                    currentSecurityBannerDismissed -> SecurityBannerState.None
-                    syncState == SyncState.Running -> {
-                        when (recoveryState) {
-                            RecoveryState.DISABLED -> SecurityBannerState.SetUpRecovery
-                            RecoveryState.INCOMPLETE -> SecurityBannerState.RecoveryKeyConfirmation
-                            RecoveryState.UNKNOWN,
-                            RecoveryState.WAITING_FOR_SYNC,
-                            RecoveryState.ENABLED -> SecurityBannerState.None
-                        }
-                    }
-                    needsSlidingSyncMigration -> SecurityBannerState.NeedsNativeSlidingSyncMigration
-                    else -> SecurityBannerState.None
-                }
+                calculateBannerState(
+                    securityBannerDismissed = currentSecurityBannerDismissed,
+                    needsSlidingSyncMigration = currentNeedsSlidingSyncMigration,
+                    recoveryState = recoveryState,
+                )
             }
         }
     }
 
+    private fun calculateBannerState(
+        securityBannerDismissed: Boolean,
+        needsSlidingSyncMigration: Boolean,
+        recoveryState: RecoveryState,
+    ): SecurityBannerState {
+        if (securityBannerDismissed) {
+            return SecurityBannerState.None
+        }
+
+        when (recoveryState) {
+            RecoveryState.DISABLED -> return SecurityBannerState.SetUpRecovery
+            RecoveryState.INCOMPLETE -> return SecurityBannerState.RecoveryKeyConfirmation
+            RecoveryState.UNKNOWN,
+            RecoveryState.WAITING_FOR_SYNC,
+            RecoveryState.ENABLED -> Unit
+        }
+
+        if (needsSlidingSyncMigration) {
+            return SecurityBannerState.NeedsNativeSlidingSyncMigration
+        }
+
+        return SecurityBannerState.None
+    }
```

Comment:
(For later) `RoomListPresenter` is quite a large file, so maybe this logic could be extracted and covered by a few unit tests?